ckanext/activity/views.py

```diff
@@ -0,0 +1,729 @@
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+import logging
+
+from datetime import datetime
+from typing import Any, Optional, Union, cast
+
+from flask import Blueprint
+
+import ckan.plugins.toolkit as tk
+import ckan.model as model
+from ckan.views.group import (
+    set_org,
+    # TODO: don't use hidden funcitons
+    _get_group_dict, _get_group_template, _replace_group_org
+)
+# TODO: don't use hidden funcitons
+from ckan.views.user import _extra_template_variables
+# TODO: don't use hidden funcitons
+from ckan.views.dataset import _setup_template_variables
+
+from ckan.types import Context, Response
+from .model import Activity
+
+log = logging.getLogger(__name__)
+bp = Blueprint("activity", __name__)
+
+
+@bp.route('/dataset/<id>/resources/<resource_id>/archive/<activity_id>')
+def resource_archive(id: str, resource_id: str, activity_id: str) -> str:
+    context = cast(Context, {
+        u'model': model,
+        u'session': model.Session,
+        u'user': tk.g.user,
+        u'auth_user_obj': tk.g.userobj,
+        u'for_view': True
+    })
+
+    try:
+        package = tk.get_action(u'package_show')(context, {u'id': id})
```

Comment:
Yes, I like this idea. I'll keep the current formatting till activity-pagination PR is merged so that there will be fewer conflicts. And then I'll run all activity-plugin files through `black`